### ISSUE TITLE
[FEATURE] Horde definition "shuffle"

### ISSUE THREAD

**Issue by Xenaero** [odamex/odamex#690]
**Is your feature request related to a problem? Please describe.**
There is a high likelihood of the beginning definitions repeating in the early rounds from map to map. This does somewhat extend to later rounds as well, however is very noticeable in early rounds. This request stems from an observation made over the past couple Monday Night Horde sessions.

For instance, the definitions "Snap, Crackle, & Pop" and "You'll Die Screaming" may, by their difficulty definition, show up in early rounds, but seeing them repeated in following maps limits other difficulty comparable definitions such as "Pac Attack" from showing up. This has the result of hampering variety for both new and returning players.

**Describe the solution you'd like to see**
The chance of waves showing up in subsequent maps dropping dramatically, re-rolling the dice on picking a definition should they show up.

**Describe alternatives you've considered**
A server-controlled variable that allows for a toggle between current behavior where anything goes, to the behavior suggested in the solution above. This would let server owners curate their Horde experience, even in mappacks with relatively limited HORDEDEF selection.

**Additional context**
This could be misconstrued as forcing variety in a system that may have limited definitions per difficulty, but it runs the risk of also being seen as stale if the system lands on the same definitions for multiple maps in a row. A shuffling behavior can boost player engagement if they know something fresh has a higher chance of showing up. It's also fun.

**Comment by leximayfield** [comment]
I agree.  My idea was to basically keep a running tally of the last `<number of waves>` defines and in subsequent attempts or maps.  Just hadn't gotten around to it yet.

**Comment by electricbrass** [comment]
Added in #1579
